src/main/java/com/google/sps/app/servlets/AddUsersToEvent.java

```diff
@@ -0,0 +1,146 @@
+import com.google.cloud.datastore.Key;
+import com.google.cloud.datastore.KeyFactory;
+import com.google.cloud.datastore.Query;
+import com.google.cloud.datastore.QueryResults;
+import com.google.cloud.datastore.Value;
+import com.google.cloud.datastore.StructuredQuery.PropertyFilter;
+import com.google.datastore.v1.client.DatastoreHelper;
+import com.google.gson.Gson;
+import com.google.sps.model.UserList;
+import com.google.sps.util.DataStoreHelper;
+import com.google.sps.util.Validator;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/addUserToEvent")
+public class AddUsersToEvent extends HttpServlet {
+    
+    private static String EVENT_ID = "javascript-input-event-id";
+    private static String USERS = "text-input-user-list";
+
+    /**
+     * Reads an events userlist and adds users that aren't there yet.
+     * @param request. Containing an event ID and
+     * a comma seperated list of usernames too add.
+     * @param response
+     * @return response containing a JSON representation of the new username list
+     * and a list of usernames that failed to add, with an error message.
+     * @throws IOException
+     */
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        //get paramaters
+        String id = request.getParameter(EVENT_ID);
+        List<String> usersToAdd = cleanInput(request.getParameter(USERS));
+        
+        //get lists
+        Entity eventEntity = getEventEntity(id);
+        List<String> usernames = getAssociatedEventUsernames(eventEntity);
+        List<String> userIDs = getAssociatedEventUserIDs(eventEntity);
+
+        //add to lists
+        UserList resp = addUsersToExistingUsers(usersToAdd, usernames, userIDs);
+
+        pushUserstoEvent(eventEntity, resp);
+
+        //return response
+        Gson gson = new Gson();
+        response.setContentType("appliation/json;");
+        response.getWriter().println(gson.toJson(resp));
+    }
+
+    private List<String> getAssociatedEventUserIDs(Entity eventEntity) {
+        List<Value<String>> datastoreList = eventEntity.getList("associatedUserIDs");
+        List<String> list = DataStoreHelper.convertToStringList(datastoreList);
+        return list;
+    }
+
+    private List<String> getAssociatedEventUsernames(Entity eventEntity) {
+        List<Value<String>> usernames = eventEntity.getList("associatedUsernames");
+        List<String> list = DataStoreHelper.convertToStringList(usernames);
+        return list;
+    }
+
+    private List<String> cleanInput(String input){
+        input = input.replace(" ", "");
+        String[] strArr = input.split(",");
+        return Arrays.asList(strArr);
+    }
+
+    private UserList addUsersToExistingUsers(List<String> usersToAdd, List<String> usernames, List<String> userIDs) {
```

Comment:
Are usernames the existingUsers? If so can you call it that?

Reply:
Same with userIds.

Reply:
good idea, done